### PR TITLE
Check start <= end in isBetween

### LINKED ISSUE #994: Check start <= end in isBetween and isStrictlyBetween assertions
For the time being, it is possible to pass an end < start :(.

AssertJ should throw an `IllegalArgumentException` in that case

### PR DESCRIPTION
**Pull request by epeee**
#### Check List:
* Fixes #994 
* Unit tests : YES
* Javadoc with a code example (API only) : NA

### CODE REVIEW

**Comment by joel-costigliola** [comment]
Thanks @epeee, @PascalSchumacher or I will review that soonish.

File: src/main/java/org/assertj/core/internal/Comparables.java [hunk 1]
```diff
@@ -255,12 +256,16 @@ private boolean isLessThan(Object actual, Object other) {
...
    * @throws NullPointerException if start value is {@code null}.
    * @throws NullPointerException if end value is {@code null}.
+   * @throws IllegalArgumentException if end value is less than start value.
    */
   public <T extends Comparable<? super T>> void assertIsBetween(AssertionInfo info, T actual, T start, T end,
       boolean inclusiveStart, boolean inclusiveEnd) {
     assertNotNull(info, actual);
     checkNotNull(start, "The start range to compare actual with should not be null");
     checkNotNull(end, "The end range to compare actual with should not be null");
+    checkArgument(comparisonStrategy.isLessThanOrEqualTo(start, end),
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/internal/Comparables.java]: if we use `isStrictlyBetween` then we should not accept start == end.

Comment (epeee) [inline src/main/java/org/assertj/core/internal/Comparables.java]: @joel-costigliola thx, currently we end up in a `AssertionError` in such a case. If this change is ok I can change the implementation to also throw an `IllegalArgumentException` in this case.

**Comment by PascalSchumacher** [comment]
Thanks! :+1:

**Pull request closed without merging**